### ISSUE TITLE
[REGRESSION] Kodi v18.8 (EDL) muting de-syncs audio 8/12/20

### ISSUE THREAD

**Issue by lilgandhi1199** [xbmc/xbmc#17328]
## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Back in Kodi Version 17.0 it will mute starting at the indicated time and unmute at the end time (as it is marked precisely in the EDL file).

Here is a comparison video of the last stable build 17.1-RC1 (GIT:2017-02-22-964306f)
https://drive.google.com/file/d/1G6YFIc4Owpdt2y3hj7W3mZQrA7NZyDhc/view?usp=sharing

## Actual Behavior

The problem is that instead of giving us silence at the start of the mute time it plays the next available sound (which would be the sound, possibly in the buffer, that comes immediately after the marked end mute time)
So, it will play a seemingly random length of audio from the future which won't line up with the lips at all. And it will then silence the audio until it reaches the end of the stolen audio buffer.   <<<    Instead of unmuting at the marked end time.

Here is a simple video demonstration. Watch their lips
https://drive.google.com/file/d/1Njj79fA5fpLJuu7obMJ5F1LM68D1qBJz/view?usp=sharing

## Possible Fix

Remove the patch that came in and messed with the EDL code in V17.1
https://github.com/xbmc/xbmc/pull/11714
^^^ here

### To Reproduce

1. Download
https://drive.google.com/open?id=1YBoUXXO3CcjUNgl-eg_Otsu_vwjhwtxm
!!!Ignore log file which was on Windows 7 originally with Kodi v18.5
18.8 log on windows 10 is below >>>

~~~~~~~~~~~~~~~~~~~~
WATCH Jimmy Stewart's LIPS.
You will see him grunting repeatedly then on his last grunt you will suddenly see him talking from the future audio.
~~~~~~~~~~~~~~~~~~~~~

2...Watch it with and without the edl file in the same folder to get a feel for the dialogue. Note audio should be set to Stereo PCM through DirectSound possibly

3. Play my clip or try it on lots of other videos too that you're more familiar with and watch their lips to see when the audio is getting played ahead of time.
...Sometimes it won't play sound from the future (but I think honestly those times it was just silent in the movie where it stole the sound from)
I notice that 90% of every video does this. And it happens on 4 of my computers Windows 7 Windows 8 and Windows 10 included. I have further tested on 2 more machines with ubuntu 16.04 through 20.04
It happens on full 1080p blurays and 480p dvds and everything in between from my experience. Audio formats don't matter either.

### Debuglog

The debuglog can be found here:

[kodi.log](https://drive.google.com/file/d/1KQXuCq7K4G_8tWxTfIhE29bLXbEuGUip)
^^^Kodi V18.8 on windows 10
!!!The log in the original download is for 18.5 which can be ignored

As requested
Kodi Nightly Log 8/13/20
https://pastebin.com/CzNeTfML

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

The other person who initially brought up the most recent EDL (commercials don't skip because of comments) bug over on the forums mentioned this lingering issue (once the comments bug was fixed)
His name is "Astrocyte74" and you can see what he said here
https://forum.kodi.tv/showthread.php?tid=349685
"There is still the annoying audio sync issue for a few seconds before and after every '1' (mute) where as with kodi 17.0 (and earlier builds) the mute worked perfectly.  Should I report this somewhere?"
I have also found that v16.1 did not have this issue

EDIT:
Here is another description of his issue which sounds like mine too
"With 18.4, .edl files worked.  Though, since v17 .edl files have caused a temporary audio sync issue but it always re-adjusts / catches up.  Here is a log for 18.4 running the same movie / edl file:"
^^^But there's his log too...actually that has the previous bug in it which seems to have been fixed.
I'll ask him for a new log only showing this bug which remains which we both seem to have.

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [X] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [X] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version: 18.8

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by DaVukovic** [comment]
Hm, unfortunately I can't reproduce your issue on Linux with my own video files and your EDL file. The muting steps in at that exact time, then everything is muted until the end-time hast reached. Then I hear the normal audio of the video until the next start-time has reached and it's again muted until the end-time. 

Will download your file and see if that behaves differently

**Comment by lilgandhi1199** [comment]
I have another gentlemen over on the forums who is likely experiencing the same issue and so I messaged him to come tell us more aswell.
He's also on Windows

Here is another description of his issue which sounds like mine too
"With 18.4, .edl files worked.  Though, since v17 .edl files have caused a temporary audio sync issue but it always re-adjusts / catches up.  Here is a log for 18.4 running the same movie / edl file:"


Might not be an issue on linux

However everyone just make sure to watch their lips. Also like I said depending on the source material the audio that it steals from after the expected mute time might be silent or uninteresting so you might not notice it either.

However, Thank you for testing yourself sir DaVukovic.

**Comment by lilgandhi1199** [comment]
Let me get the log from my Windows 10 PC as well.
I'll be back later with that

**Comment by lilgandhi1199** [comment]
@fuzzard 
@the-black-eagle 

Before V19 comes out I know one of you fixed a recent EDL file bug and wonder if you can tackle this one too?

I'm ready to get more logs and do more testing for you when you're ready!

**Comment by lilgandhi1199** [comment]
@FernetMenta Hey I heard from the-black-eagle that you might be the one to fix this if you can figure it out due to it being an audio bug.
He also said Fritsch is retired but would also know.

Maybe I'll install the nightly version and see if you guys ended up fixing it already maybe with your last review of the EDL code!
Where would I find that though?

**Comment by lilgandhi1199** [comment]
Alright. Forget about nightlies.
I got the log from Version 17.0 when everything worked perfectly
[kodi-17.0-pre_bug.log](https://github.com/xbmc/xbmc/files/4196238/kodi-17.0-pre_bug.log)
and then after the bug popped up the log from 17.1
[kodi-17.1-post_bug.log](https://github.com/xbmc/xbmc/files/4196241/kodi-17.1-post_bug.log)

I noticed in the changelog someone was in the EDL code.

> Added support for navigation controls between EDL (commercial skipping) during PVR recodings playback

And it referenced this github pull
https://github.com/xbmc/xbmc/pull/10064

**Comment by lilgandhi1199** [comment]
@DaVukovic 
I think I figured it out! ^^^
Can you provide the nightlies exe install file from 17.0 before 17.1 came out (specifically the day before that request was merged into the project and the day after it was implemented?)

Then if I can prove it is infact his code by testing it this way we could just disable his code in the current version of Kodi and watch the flag fly in the breeze of victory!

**Comment by lilgandhi1199** [comment]
`19:00:13.159 T:1560   DEBUG: CDVDAudio::Resume - resume audio stream
19:00:13.159 T:6604   DEBUG: ActiveAE - start sync of audio stream
19:00:13.161 T:5244   DEBUG: CVideoPlayerVideo - CDVDMsg::GENERAL_RESYNC(-100000.000000)
19:00:13.346 T:6960   DEBUG: CAESinkDirectSound::CheckPlayStatus: Resuming Playback
19:00:13.347 T:6604   DEBUG: ActiveAE::SyncStream - average error of 113.875783, start adjusting
19:00:13.347 T:6604   DEBUG: ActiveAE::SyncStream - average error 23.875783 below threshold of 30.000000
19:00:13.668 T:5244  NOTICE: CDVDVideoCodecFFmpeg::CDropControl: calculated diff time: 53090
19:00:14.352 T:6604   DEBUG: ActiveAE::SyncStream - average error -188.926587 above threshold of 100.000000
19:00:14.352 T:6604   DEBUG: ActiveAE::SyncStream - average error -23.926587 below threshold of 30.000000
19:00:14.583 T:5660    INFO: XCURL::DllLibCurlGlobal::CheckIdle - Closing session to http://mirrors.kodi.tv (easy=0D1BCAF0, multi=051C5328)
19:00:15.387 T:1560   DEBUG: CDVDClock::ErrorAdjust - CVideoPlayerAudio::OutputPacket - error:-25425.385531, adjusted:-25425.385531
19:00:18.316 T:5244   DEBUG: CPullupCorrection: detected pattern of length 1: 41708.33, frameduration: 41708.333333
19:00:21.866 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:21.866 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:21.866 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:21.866 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:21.866 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.364 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.364 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.364 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.364 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.365 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.397 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.397 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.397 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.398 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.398 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.431 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.431 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.431 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.431 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.447 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.464 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.464 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.464 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.464 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.464 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.497 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.497 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.498 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.498 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.498 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.530 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.531 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.531 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.531 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.531 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.564 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.564 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.564 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.564 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.564 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.581 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.581 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.581 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.581 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.597 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.614 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.615 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.615 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.616 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.631 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.664 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.666 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.666 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.666 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.667 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.680 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.681 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.681 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.681 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.697 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.714 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.714 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.714 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.714 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.731 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.747 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.747 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.748 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.748 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.748 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.780 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.780 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.781 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.781 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.781 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.814 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.814 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.814 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.814 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.814 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.847 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.847 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.847 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.847 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.848 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.880 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.881 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.881 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.881 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.882 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.914 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.914 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.914 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.914 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.914 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.947 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.947 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.947 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.947 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.948 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.964 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:22.964 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:22.964 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:22.964 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:22.981 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.014 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.014 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.014 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.014 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.014 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.030 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.030 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.031 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.031 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.047 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.064 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.064 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.064 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.064 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.081 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.097 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.097 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.097 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.097 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.098 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.130 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.131 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.131 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.131 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.147 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.164 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.164 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.164 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.164 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.181 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.197 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.197 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.197 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.197 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.197 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.230 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.230 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.230 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.231 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.231 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.264 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.264 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.266 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.266 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.266 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.297 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.297 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.297 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.297 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.297 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.330 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.330 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.331 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.331 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.331 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.364 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.364 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.364 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.364 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.364 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.397 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.426 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.426 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.426 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.427 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.427 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.427 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.427 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.427 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.427 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.447 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.447 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.447 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.447 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.464 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.480 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.480 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.481 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.481 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.481 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.513 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.514 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.514 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.514 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.530 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.547 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.547 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.547 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.548 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.548 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.580 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.580 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.581 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.581 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.581 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.613 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.614 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.614 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.614 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.630 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.647 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.647 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.647 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.647 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.647 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.680 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.680 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.680 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.680 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.681 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.713 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.714 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.714 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.714 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.714 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.747 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.747 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.748 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.748 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.748 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.763 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.763 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.764 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.764 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.780 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.797 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.797 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.797 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.797 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.813 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.830 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.830 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.830 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.831 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.847 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.864 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.864 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.864 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.864 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.881 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.897 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.897 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.897 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.897 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.913 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.930 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.930 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.930 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.930 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.930 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.963 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.964 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.964 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.964 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.964 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.997 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:23.997 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:23.997 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:23.997 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:23.997 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:24.030 T:5660   DEBUG: WinEventsWin32.cpp: APPCOMMAND 10
19:00:24.030 T:5660   DEBUG: CApplication::OnAppCommand: unknown appcommand 10
19:00:24.030 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:24.030 T:5660   DEBUG: CInputManager::OnKey: 0 (0xf200, obc-61697) pressed, action is
19:00:24.030 T:5660   DEBUG: Keyboard: scancode: 0x30, sym: 0x0000, unicode: 0x0000, modifier: 0x0
19:00:36.423 T:1560   DEBUG: CVideoPlayerAudio - CDVDMsg::AUDIO_SILENCE(23901000.000000, 1)
19:00:40.039 T:1560   DEBUG: CVideoPlayerAudio - CDVDMsg::AUDIO_SILENCE(27517000.000000, 0)
19:00:43.181 T:4044   DEBUG: Thread JobWorker 4044 terminating (autodelete)
19:00:43.199 T:4456   DEBUG: Thread JobWorker 4456 terminating (autodelete)
19:00:43.200 T:4608   DEBUG: Thread JobWorker 4608 terminating (autodelete)
19:00:44.975 T:1560   DEBUG: CVideoPlayerAudio - CDVDMsg::AUDIO_SILENCE(32445000.000000, 1)
19:00:50.406 T:1560   DEBUG: CVideoPlayerAudio - CDVDMsg::AUDIO_SILENCE(37885000.000000, 0)
19:00:59.242 T:1560   DEBUG: CVideoPlayerAudio - CDVDMsg::GENERAL_EOF
19:00:59.639 T:1560  NOTICE: CVideoPlayerAudio::Process - stream stalled
19:00:59.863 T:4300  NOTICE: CVideoPlayer::OnExit()
19:00:59.863 T:4300  NOTICE: VideoPlayer: eof, waiting for queues to empty
19:00:59.863 T:4300  NOTICE: Closing stream player 1
19:00:59.863 T:4300  NOTICE: CDVDMessageQueue(audio)::WaitUntilEmpty
19:00:59.863 T:4300  NOTICE: Waiting for audio thread to exit
19:00:59.863 T:1560   DEBUG: CVideoPlayerAudio - CDVDMsg::GENERAL_SYNCHRONIZE
19:00:59.863 T:1560  NOTICE: thread end: CVideoPlayerAudio::OnExit()
19:00:59.863 T:4300  NOTICE: Closing audio device
19:00:59.863 T:1560   DEBUG: Thread VideoPlayerAudio 1560 terminating
19:00:59.864 T:6604   DEBUG: CActiveAE::DiscardStream - audio stream deleted
19:00:59.864 T:6604   DEBUG: CActiveAE::ClearDiscardedBuffers - buffer pool deleted
19:00:59.865 T:4300   DEBUG: Previous line repeats 2 times.
19:00:59.865 T:4300  NOTICE: Deleting audio codec
19:00:59.865 T:4300  NOTICE: Closing stream player 2
19:00:59.865 T:4300  NOTICE: CDVDMessageQueue(video)::WaitUntilEmpty
19:00:59.865 T:5244   DEBUG: CPullupCorrection: pattern lost on diff 125341.666667, number of losses 1
19:00:59.867 T:5244   DEBUG: CVideoPlayerVideo - CDVDMsg::GENERAL_SYNCHRONIZE
19:00:59.868 T:5244   ERROR: Got MSGQ_ABORT or MSGO_IS_ERROR return true
19:00:59.868 T:5244  NOTICE: thread end: video_thread`
^^^ Here is the pre bug log from initialization from start to finish which clearly shows the intended mutes towards the end under "CDVDMsg::AUDIO_SILENCE"



`18:53:43.887 T:340   DEBUG: CDVDAudio::Resume - resume audio stream
18:53:43.887 T:2816   DEBUG: ActiveAE - start sync of audio stream
18:53:43.891 T:5360   DEBUG: CVideoPlayerVideo - CDVDMsg::GENERAL_RESYNC(-100000.000000)
18:53:44.047 T:5312   DEBUG: CAESinkDirectSound::CheckPlayStatus: Resuming Playback
18:53:44.048 T:2816   DEBUG: ActiveAE::SyncStream - average error of 115.680305, start adjusting
18:53:44.048 T:2816   DEBUG: ActiveAE::SyncStream - average error 25.680305 below threshold of 30.000000
18:53:44.389 T:5360  NOTICE: CDVDVideoCodecFFmpeg::CDropControl: calculated diff time: 53090
18:53:45.054 T:2816   DEBUG: ActiveAE::SyncStream - average error -162.555705 above threshold of 100.000000
18:53:45.054 T:2816   DEBUG: ActiveAE::SyncStream - average error -27.555705 below threshold of 30.000000
18:53:46.089 T:340   DEBUG: CDVDClock::ErrorAdjust - CVideoPlayerAudio::OutputPacket - error:-29094.018930, adjusted:-29094.018930
18:53:49.053 T:5360   DEBUG: CPullupCorrection: detected pattern of length 1: 41708.33, frameduration: 41708.333333
18:54:08.175 T:2816   DEBUG: ActiveAE::SyncStream - average error 2267.233947 above threshold of 100.000000
18:54:09.968 T:340   ERROR: CDVDAudio::AddPacketsRenderer - timeout adding data to renderer
18:54:11.805 T:2816   ERROR: Previous line repeats 1 times.
18:54:11.805 T:2816   DEBUG: ActiveAE::SyncStream - average error 16.566780, last average error: 2267.233947
18:54:11.910 T:2816   DEBUG: ActiveAE::SyncStream - average error of -28.430362, start adjusting
18:54:11.910 T:2816   DEBUG: ActiveAE::SyncStream - average error -13.430362 below threshold of 30.000000
18:54:12.945 T:340   DEBUG: CDVDClock::ErrorAdjust - CVideoPlayerAudio::OutputPacket - error:16939.173684, adjusted:16939.173684
18:54:13.813 T:6532   DEBUG: Thread JobWorker 6532 terminating (autodelete)
18:54:13.981 T:340   DEBUG: CDVDClock::ErrorAdjust - CVideoPlayerAudio::OutputPacket - error:13176.668329, adjusted:13176.668329
18:54:16.937 T:2816   DEBUG: ActiveAE::SyncStream - average error 4708.749623 above threshold of 100.000000
18:54:18.731 T:340   ERROR: CDVDAudio::AddPacketsRenderer - timeout adding data to renderer
18:54:22.372 T:2816   ERROR: Previous line repeats 2 times.
18:54:22.372 T:2816   DEBUG: ActiveAE::SyncStream - average error 24.684830, last average error: 4708.749623
18:54:22.477 T:2816   DEBUG: ActiveAE::SyncStream - average error of -20.316072, start adjusting
18:54:22.477 T:2816   DEBUG: ActiveAE::SyncStream - average error -5.316072 below threshold of 30.000000
18:54:23.512 T:340   DEBUG: CDVDClock::ErrorAdjust - CVideoPlayerAudio::OutputPacket - error:55406.093775, adjusted:55406.093775
18:54:24.547 T:340   DEBUG: CDVDClock::ErrorAdjust - CVideoPlayerAudio::OutputPacket - error:26972.030700, adjusted:26972.030700
18:54:29.858 T:340   DEBUG: CVideoPlayerAudio - CDVDMsg::GENERAL_EOF
18:54:30.250 T:340  NOTICE: CVideoPlayerAudio::Process - stream stalled
18:54:30.461 T:2940  NOTICE: CVideoPlayer::OnExit()
18:54:30.461 T:2940  NOTICE: VideoPlayer: eof, waiting for queues to empty
18:54:30.461 T:2940  NOTICE: Closing stream player 1
18:54:30.461 T:2940  NOTICE: CDVDMessageQueue(audio)::WaitUntilEmpty
18:54:30.461 T:340   DEBUG: CVideoPlayerAudio - CDVDMsg::GENERAL_SYNCHRONIZE
18:54:30.461 T:2940  NOTICE: Waiting for audio thread to exit
18:54:30.461 T:340   ERROR: Got MSGQ_ABORT or MSGO_IS_ERROR return true
18:54:30.461 T:340  NOTICE: thread end: CVideoPlayerAudio::OnExit()
18:54:30.461 T:2940  NOTICE: Closing audio device
18:54:30.461 T:340   DEBUG: Thread VideoPlayerAudio 340 terminating
18:54:30.462 T:2816   DEBUG: CActiveAE::DiscardStream - audio stream deleted
18:54:30.462 T:2816   DEBUG: CActiveAE::ClearDiscardedBuffers - buffer pool deleted
18:54:30.463 T:2940   DEBUG: Previous line repeats 2 times.
18:54:30.463 T:2940  NOTICE: Deleting audio codec
18:54:30.463 T:2940  NOTICE: Closing stream player 2
18:54:30.463 T:2940  NOTICE: CDVDMessageQueue(video)::WaitUntilEmpty
18:54:30.463 T:5360   DEBUG: CPullupCorrection: pattern lost on diff 125341.666667, number of losses 1
18:54:30.488 T:5360   DEBUG: CVideoPlayerVideo - CDVDMsg::GENERAL_SYNCHRONIZE
18:54:30.491 T:2940  NOTICE: waiting for video thread to exit
18:54:30.491 T:5360   ERROR: Got MSGQ_ABORT or MSGO_IS_ERROR return true
18:54:30.491 T:5360  NOTICE: thread end: video_thread`
^^^ Then once the bug exists it doesn't even mention the intended "CDVDMsg::AUDIO_SILENCE" anywhere! Plus it knows it's desyncing I think because each mute ends up triggering this in the log.

"CDVDAudio::AddPacketsRenderer - timeout adding data to renderer"
In the above log you'll see two instances of it (and remember this is the edl file I'm using)
00:00:23.897 00:00:27.500 1
00:00:32.427 00:00:37.862 1

So I increased the number of mutes from 2 to 6 with the following edl file...
00:00:23.897 00:00:27.500 1
00:00:32.427 00:00:37.862 1
00:00:38.000 00:00:39.000 1
00:00:40.000 00:00:41.000 1
00:00:42.000 00:00:43.000 1
00:00:44.000 00:00:45.000 1

So here is that same section of the log in V17.1 with the ^^^ 6 mutes
`20:40:01.568 T:3932   DEBUG: CDVDAudio::Resume - resume audio stream
20:40:01.568 T:6380   DEBUG: ActiveAE - start sync of audio stream
20:40:01.743 T:4668   DEBUG: CAESinkDirectSound::CheckPlayStatus: Resuming Playback
20:40:01.744 T:6380   DEBUG: ActiveAE::SyncStream - average error of 114.720887, start adjusting
20:40:01.744 T:6380   DEBUG: ActiveAE::SyncStream - average error 24.720887 below threshold of 30.000000
20:40:02.075 T:6572  NOTICE: CDVDVideoCodecFFmpeg::CDropControl: calculated diff time: 53090
20:40:02.750 T:6380   DEBUG: ActiveAE::SyncStream - average error -176.869588 above threshold of 100.000000
20:40:02.750 T:6380   DEBUG: ActiveAE::SyncStream - average error -26.869588 below threshold of 30.000000
20:40:03.802 T:3932   DEBUG: CDVDClock::ErrorAdjust - CVideoPlayerAudio::OutputPacket - error:-28355.651530, adjusted:-28355.651530
20:40:06.738 T:6572   DEBUG: CPullupCorrection: detected pattern of length 1: 41708.33, frameduration: 41708.333333
20:40:07.304 T:6332    INFO: Python, unloading python shared library because no scripts are running anymore
20:40:12.968 T:6332    INFO: XCURL::DllLibCurlGlobal::CheckIdle - Closing session to http://mirrors.kodi.tv (easy=0E2A9638, multi=04C440B0)
20:40:25.869 T:6380   DEBUG: ActiveAE::SyncStream - average error 2321.207007 above threshold of 100.000000
20:40:27.662 T:3932   ERROR: CDVDAudio::AddPacketsRenderer - timeout adding data to renderer
20:40:29.499 T:6380   ERROR: Previous line repeats 1 times.
20:40:29.499 T:6380   DEBUG: ActiveAE::SyncStream - average error 16.580076, last average error: 2321.207007
20:40:29.604 T:6380   DEBUG: ActiveAE::SyncStream - average error of -28.432660, start adjusting
20:40:29.604 T:6380   DEBUG: ActiveAE::SyncStream - average error -13.432660 below threshold of 30.000000
20:40:30.639 T:3932   DEBUG: CDVDClock::ErrorAdjust - CVideoPlayerAudio::OutputPacket - error:28938.223428, adjusted:28938.223428
20:40:31.548 T:4904   DEBUG: Thread JobWorker 4904 terminating (autodelete)
20:40:31.549 T:4644   DEBUG: Thread JobWorker 4644 terminating (autodelete)
20:40:31.549 T:5308   DEBUG: Thread JobWorker 5308 terminating (autodelete)
20:40:31.674 T:3932   DEBUG: CDVDClock::ErrorAdjust - CVideoPlayerAudio::OutputPacket - error:18531.519670, adjusted:18531.519670
20:40:34.629 T:6380   DEBUG: ActiveAE::SyncStream - average error 5640.681592 above threshold of 100.000000
20:40:36.414 T:3932   ERROR: CDVDAudio::AddPacketsRenderer - timeout adding data to renderer
20:40:41.067 T:6380   ERROR: Previous line repeats 2 times.
20:40:41.067 T:6380   DEBUG: ActiveAE::SyncStream - average error 26.344889, last average error: 5640.681592
20:40:41.172 T:6380   DEBUG: ActiveAE::SyncStream - average error of -18.658632, start adjusting
20:40:41.172 T:6380   DEBUG: ActiveAE::SyncStream - average error -3.658632 below threshold of 30.000000
20:40:42.177 T:6380   DEBUG: ActiveAE::SyncStream - average error 891.559258 above threshold of 100.000000
20:40:43.032 T:6380   DEBUG: ActiveAE::SyncStream - average error 21.559258, last average error: 891.559258
20:40:43.137 T:6380   DEBUG: ActiveAE::SyncStream - average error of 472.830278, start adjusting
20:40:44.322 T:6380   DEBUG: ActiveAE::SyncStream - average error 26.418226, last average error: 472.830278
20:40:44.427 T:6380   DEBUG: ActiveAE::SyncStream - average error of -18.573813, start adjusting
20:40:44.435 T:6380   DEBUG: ActiveAE::SyncStream - average error -3.573813 below threshold of 30.000000
20:40:45.447 T:6380   DEBUG: ActiveAE::SyncStream - average error 125.828871 above threshold of 100.000000
20:40:45.537 T:6380   DEBUG: ActiveAE::SyncStream - average error 20.828871 below threshold of 30.000000
20:40:46.542 T:6380   DEBUG: ActiveAE::SyncStream - average error 883.466993 above threshold of 100.000000
20:40:47.382 T:6380   DEBUG: ActiveAE::SyncStream - average error 28.466993, last average error: 883.466993
20:40:47.487 T:6380   DEBUG: ActiveAE::SyncStream - average error of 28.493401, start adjusting
20:40:47.487 T:6380   DEBUG: ActiveAE::SyncStream - average error 13.493401 below threshold of 30.000000
20:40:47.592 T:3932   DEBUG: CVideoPlayerAudio - CDVDMsg::GENERAL_EOF
20:40:47.989 T:3932  NOTICE: CVideoPlayerAudio::Process - stream stalled
20:40:48.229 T:4280  NOTICE: CVideoPlayer::OnExit()
20:40:48.229 T:4280  NOTICE: VideoPlayer: eof, waiting for queues to empty
20:40:48.229 T:4280  NOTICE: Closing stream player 1
20:40:48.229 T:4280  NOTICE: CDVDMessageQueue(audio)::WaitUntilEmpty
20:40:48.229 T:4280  NOTICE: Waiting for audio thread to exit
20:40:48.229 T:3932   DEBUG: CVideoPlayerAudio - CDVDMsg::GENERAL_SYNCHRONIZE
20:40:48.229 T:3932  NOTICE: thread end: CVideoPlayerAudio::OnExit()
20:40:48.229 T:4280  NOTICE: Closing audio device
20:40:48.229 T:3932   DEBUG: Thread VideoPlayerAudio 3932 terminating
20:40:48.231 T:6380   DEBUG: CActiveAE::DiscardStream - audio stream deleted
20:40:48.231 T:6380   DEBUG: CActiveAE::ClearDiscardedBuffers - buffer pool deleted
20:40:48.232 T:4280   DEBUG: Previous line repeats 2 times.
20:40:48.232 T:4280  NOTICE: Deleting audio codec
20:40:48.232 T:4280  NOTICE: Closing stream player 2
20:40:48.232 T:4280  NOTICE: CDVDMessageQueue(video)::WaitUntilEmpty
20:40:48.232 T:6572   DEBUG: CPullupCorrection: pattern lost on diff 125341.666667, number of losses 1
20:40:48.232 T:6572   DEBUG: CVideoPlayerVideo - CDVDMsg::GENERAL_SYNCHRONIZE
20:40:48.232 T:4280  NOTICE: waiting for video thread to exit
20:40:48.232 T:6572   ERROR: Got MSGQ_ABORT or MSGO_IS_ERROR return true
20:40:48.232 T:6572  NOTICE: thread end: video_thread`

Though the timeout renderer hasn't increased proportionally this "ActiveAE::SyncStream" still seems to be getting triggered a lot!

**Comment by lilgandhi1199** [comment]
Please! Someone reverse this before everything jumps to windows 10 and my chance to have proper muting AND v18 is completely lost.

**Comment by phunkyfish** [comment]
@lilgandhi1199 can you remove your log paste from this issue and use a paste site or a gist instead?

**Comment by lilgandhi1199** [comment]
Yes let me get right on that!
I have tried to compile kodi on my ubuntu machine to fix this myself in the past weeks and could not get it going after following the step by step guide. Any interest in this case is really greatly apreciated!!

Thank you as well to @techymomof6 adding his experience

> @phunkyfish
> 
> I tried it on the Matrix last night, same results as described by the-black-eagle.
> 
> ".edl file doesn't correctly skip or mute."
> 
> The audio is out of sync until it catches up to video after mute
> 
> I find if I have multiple skips together like for inappropriate scenes for kids, it will sometimes skip whole chunks. For example if I have 3 skips in a row doing the same time in a movie it may skip from the beginning of the first time stamp to the end of the 3rd stamp, instead of playing /skipping the stamp in between. Also it has a buffering time while it skips with visible % wheel while it catches up. The skip/mute only works perfectly in 16.1 from what I've tested.
> 
> Hope this makes sense, I don't have programming experience.

**Comment by phunkyfish** [comment]
@djp952 do you have much experience with EDL files and Video Player by any chance? I know you had some some stuff with audio in demuxer so just asking.

**Comment by lilgandhi1199** [comment]
I also emailed @FernetMenta and he couldn't tell me what his original patch to the code inbetween Kodi V17.0 and 17.1 was really at all. Not willing to reverse fix it either I presume.

**Comment by lilgandhi1199** [comment]
@DaVukovic The same crap is happening on Ubuntu 20.04 newest Kodi...Ubuntu 16.04...Ubuntu 18.04
I just don't think he knew what he was supposed to be looking for.

I thought it would be obvious if you were watching their lips you'd see it was playing audio from a completely different part of the video.

**Comment by techymomof6** [comment]
@lilgandhi1199 Per your request:

Test .edl Kodi v17.0:
The mute feature works as it should
The skip feature completely freezes the movie, unable to move on

Test .edl Kodi v17.1:
(🧐 Can't seem to find a link anywhere to download this version)

At least v17 has the new layout, if only it would work with one of those releases.

**Comment by djp952** [comment]
> @djp952 do you have much experience with EDL files and Video Player by any chance? I know you had some some stuff with audio in demuxer so just asking.

Sadly no, I'm sorry @phunkyfish.  My PVR supports EDL and I know my users have been happy with it since it's return in recent Leia versions.  That's all MPEG-TS stuff, though, which might have different concerns.  They also seem to only use it for skipping commercials.

For what it's worth, in a quick look at some of the log files here I see a semi-correlation with something I've been wanting to report, but I'm not smart enough to offer any via suggestions about them.  In the event it may be a clue that helps (which I doubt), I know of problems with audio sync in a couple cases (again, I've not reported these formally): _Realtime_ MPEG-TS A/V streams where the audio bitrate is higher than the video bitrate, and _Realtime_ audio only PCM streams where the PTS falls behind realtime.  In both cases, VideoPlayer tries to resync the audio, but it can't since it's impossible to do so for something truly realtime.  It's not possible to read into the future.  In the low video bitrate case it will manifest as a regular buffer/mute/play/buffer/mute/play, in the PTS falling behind case it just can't resolve the problem and you have to give up.

Again, I doubt one thing has anything to do with the other, but I do believe there are some possible issues with audio sync logic lurking somewhere, and the parts of code being blamed are pretty much the same ones I've been looking at to see if I could offer any reasonable solutions to my problems.  I honestly don't think they would apply to streams where Kodi can find the PTS it's looking for.

If this starts leading back to anything I may have touched, please let me know, I'd bend over backwards to try and make it right!

Wow, long comment that added little value :) I gotta be me, right?

**Comment by phunkyfish** [comment]
> **edit:** Maybe something with the MUTE function that isn't tracking the PTS for the dumped audio? When it tries to fire back up the PTS is so far off from the video that bad things happen? I do see pretty large sync errors in the posted log. 20 seconds, 30 seconds, etc. VideoPlayer is above my paygrade.

Ya, me too!

**Comment by phunkyfish** [comment]
In my head muting seems to be a simple concept but I'm assuming it's just not passing the audio to the audio device if the current frame is muted. 

@FernetMenta, could use your guidance on how this should be done before trying to tackle it.

**Comment by djp952** [comment]
FWIW, I went ahead and hooked up some EDL via my PVR addon, and there is definitely something wonky with MUTE.  For me and a recorded MPEG-TS file with EDL, introduction of MUTE actually causes video playback to stop and momentarily buffer repeatedly, similar (but not exactly the same) to what I see with the realtime streams where the audio bitrate exceeds that of the video bitrate.

I do not think I can be of significant help to fix this, but I can definitely corroborate the report that something is, as I so eloquently put it above, 'wonky'.  I have some free time tonight, I'll build a Leia Debug build and see if I can perhaps assist with narrowing it down a bit.

**Comment by phunkyfish** [comment]
Thanks for any assistance you can provide @djp952 😉

**Comment by FernetMenta** [comment]
> In my head muting seems to be a simple concept but I'm assuming it's just not passing the audio to the audio device if the current frame is muted.
> 
> @FernetMenta, could use your guidance on how this should be done before trying to tackle it.

do you have a debug log created with master branch? (on vacation with phone and tablet only)

**Comment by lilgandhi1199** [comment]
@FernetMenta Is the master branch technically 19.0 beta or whatever?
Or are you referring to 18.8? Because I've always uploaded the current branch under debuglog.

**Comment by fritsch** [comment]
Master branch is the future v19. You can find the nightly builds here: http://mirrors.kodi.tv/nightlies/windows/win64/master/

Please use pastebin.com for your logfile.

**Comment by lilgandhi1199** [comment]
Alright no problem. I forgot about pastebin. Be right back.

**Comment by lilgandhi1199** [comment]
Kodi Nightly Log 8/13/20
https://pastebin.com/CzNeTfML

**Comment by djp952** [comment]
Hi @phunkyfish, I got as far as determining that the action of dropping the audio packet without processing it is the 'root cause', per-se.  By not processing the packet the DTS/PTS information and ffmpeg voodoo (that I can't debug) never take place, which ultimately leads to VideoPlayer thinking the audio stream stalled. Things start to fall apart here (Leia codebase):

https://github.com/xbmc/xbmc/blob/0e37911544f0c2baa7eb73680a6cee9181928053/xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp#L390-L396

Krypton appears to handle a dropped audio packet in what has now become CVideoPlayerAudio::ProcessDecoderOutput(), one possibility might be to pass the drop flag to that function and handle it the same way as opposed to just dropping it by ignoring it.  I haven't tried anything like that yet, it only occurred to me while diffing Krypton and Leia.

Anything I tried so far was, as expected, a colossal failure, other than not dropping the packet, which worked fine (but won't be muted).

I do hope this helps in some way.

**Comment by phunkyfish** [comment]
Thanks @djp952. Your findings make sense and if it’s the manner in which the audio packets are being dropped is the issue hopefully that will us a bit closer to the answer.

**Comment by FernetMenta** [comment]
You are wrong. VPA and AE doesn't require audio packets to be processed. That's why I dropped rhis old crap. Nevertheless , VPA should not signal stalled after having dropped packets.

**Comment by lilgandhi1199** [comment]
@enen92 I tested the one you linked me directly as well as this one I saw on the master
KodiSetup-20211116-3f0508c0-master-x64

These are all default settings Windows 10 fresh install and cleared profile folder. (No audio passthrough)

Both failed in the same respect that they always fail. The audio jumps forward to play during the beginning of the mute time stealing from the end time mute. Pushing it out further then it was supposed to end and jumbling dialogue. It should be obvious as I have made a video demonstration of it with the same test clip and EDL file I linked in my original thread.

https://drive.google.com/file/d/1Njj79fA5fpLJuu7obMJ5F1LM68D1qBJz/view?usp=sharing



It does this with all video files I have ever tested. There is nothing special about mine but it is very good at demonstrating the problem.

**Comment by lilgandhi1199** [comment]
Here is a video of the last stable version
https://drive.google.com/file/d/1G6YFIc4Owpdt2y3hj7W3mZQrA7NZyDhc/view?usp=sharing

17.1-RC1 (GIT:2017-02-22-964306f)

**Comment by lilgandhi1199** [comment]
@enen92 im sorry I just didn't edit my last post to indicate you don't need to reopen.it. I was leaving it there for posterity.
I think it's fixed atleast in the link you gave me.
thanks again
